### PR TITLE
[frontend] feat: add edit button

### LINKED ISSUE #21: [Frontend] Edit Questions
- Enable users to edit questions. 
- Implement edit buttons. 
- Create modals/ forms to edit the question.

### PR DESCRIPTION
**Pull request by juslam19**
- Also made some aesthetic changes to the question list

### CODE REVIEW

File: frontend/src/components/questions/QuestionEditIconButton.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+}: QuestionEditIconButtonProps) => {
+  const navigate = useNavigate();
+  return (
+    <Tooltip label={`Edit Question ${questionId}: ${title}`}>
+      <IconButton
+        colorScheme="yellow"
+        aria-label="Edit Question"
+        value={questionId}
+        onClick={() => {
+          navigate(`/question/edit/${questionId}`);
```
Comment (seelengxd) [inline frontend/src/components/questions/QuestionEditIconButton.tsx]: url changed in #27 

as for the aesthetic changes - i'll leave it to someone else to approve :P

**Comment by chloeelim** [comment]
Thanks for working on this! This is just a minor nit but I personally think it might be better to remove the colors on the buttons in the question list as it feels a little too overwhelming (too much cognitive overload, attention goes to the button column instead of the 'more important' data)- this might be a personal preference though, so feel free to disagree! :)

|No color|Color|
|----|----|
| Screenshot 2023-09-13 at 3 25 51 AM |  Screenshot 2023-09-13 at 3 26 28 AM    |

**Comment by sltsheryl** [comment]
> Thanks for working on this! This is just a minor nit but I personally think it might be better to remove the colors on the buttons in the question list as it feels a little too overwhelming (too much cognitive overload, attention goes to the button column instead of the 'more important' data)- this might be a personal preference though, so feel free to disagree! :)
> 
> No color	Color
> Screenshot 2023-09-13 at 3 25 51 AM	Screenshot 2023-09-13 at 3 26 28 AM

we shd also consider if the delete button (a bin icon) is beside, it’s abit too many things in the table. should we make the edit in the individual question page or it’s okay to have three buttons in each row

**Comment by juslam19** [comment]
Changed button colours to gray, except arrow which is blue

File: frontend/src/components/questions/QuestionViewIconButton copy.tsx
Comment (seelengxd) [inline frontend/src/components/questions/QuestionViewIconButton copy.tsx:1]: name change causing code to not compile

**Review by seelengxd — DISMISSED** [review]

File: frontend/src/components/questions/QuestionViewIconButton copy.tsx
Comment (juslam19) [reply frontend/src/components/questions/QuestionViewIconButton copy.tsx:1]: Can you screenshot the errors? I do not have these issues.

Comment (seelengxd) [reply frontend/src/components/questions/QuestionViewIconButton copy.tsx:1]: hmm maybe its something wrong on my end let me check my bad

image

Comment (seelengxd) [reply frontend/src/components/questions/QuestionViewIconButton copy.tsx:1]: ok nvm fixed 👍

**Comment by sltsheryl** [comment]
@chloeelim ^what do you think abt the colors 🤔

**Review by sltsheryl — APPROVED** [review]
- i changed the position of edit button to the individual question instead (we should only show this to admins) 
- restyled the delete and view buttons

# Initial
Screenshot 2023-09-27 at 5 03 27 PM

# Updated
Screenshot 2023-09-27 at 5 26 53 PM

Screenshot 2023-09-27 at 5 26 58 PM

**Pull request merged**